### PR TITLE
Fixed Microsoft.Maui.Controls 8.0.90 (Latest stable) Ruins Entry Controls

### LINKED ISSUE #24783: Microsoft.Maui.Controls 8.0.90 (Latest stable) Ruins Entry Controls
### Description

If you update the Maui Controls NuGet to the latest stable (currently 8.0.90), all Entry controls become very narrow and nothing will fix it apart from downgrading, I just went back to 8.0.72 and it is fine again.

This appears to only affect Windows. Not IOS or Android.

### Steps to Reproduce

1. Create a new MAUI project
2. Add an Entry control to the home page xaml.
3. Launch and see that it looks fine
4. Upgrade the Maui Controls NuGet to the latest stable 8.0.90
5. Launch project again and see that the Entry looks awful now.

### Link to public reproduction project repository

_No response_

### Version with bug

8.0.90

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

8.0.72

### Affected platforms

Windows

### Affected platform versions

Windows

### Did you find any workaround?

Downgrading the NuGet

### Relevant log output

_No response_

### LINKED ISSUE #25038: MAUI Entry in Windows always shows ClearButton if initially hidden and shown even if ClearButtonVisibility set to 'Never'
### Description

If you have an <Entry /> cell on a form and it's initially set as visibility "False", when it it shown (e.g. set to "True"), the ClearButton is always shown, even if "ClearButtonVisibility" is set to "Never". This seems to happen even if you programatically set ClearButtonVisilibility to 'Never' in the code.

The issue does not happen with an <Editor /> cell, so I'm assuming it is a bug.

This happens in Windows - I haven't tested it with other platforms.

### Steps to Reproduce

See description.

### Link to public reproduction project repository

_No response_

### Version with bug

9.0.0-rc.1.24453.9

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

_No response_

### Did you find any workaround?

No.

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by NirmalKumarYuvaraj**
### Root Cause
The TextBox style applied resulted in an incorrect column count, causing the deleteButton column to shift and misalign the layout.
 
### Description of Change
The code was updated to correctly identify the deleteButton column, setting its width to 0 when the clearButton is not visible and to Auto when it is visible to ensure proper layout.

 
Validated the behaviour in the following platforms
 
- [x] Android
- [x] Windows
- [x] iOS
- [x] Mac
 

### Fixes

fixes #24783
fixes #25038

### Output 
**Before**

Before Fix

**After**

After Fix

### CODE REVIEW

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue25038.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+		public override string Issue => "MAUI Entry in Windows always shows ClearButton if initially hidden and shown even if ClearButtonVisibility set to 'Never'";
+
+		[Test]
+		[Category(UITestCategories.Entry)]
+		[FailsOnMac]
+		public void VerifyEntryClearButtonVisibility()
+		{
+			App.WaitForElement("button");
+			App.Tap("button");
+			VerifyScreenshot();
```
Comment (jsuarezruiz) [inline src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue25038.cs:22]: Triggered the build, it should generate the reference snapshot for this test.

Comment (jsuarezruiz) [reply src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue25038.cs:22]: Ok, already added the test snapshot.

**Review by Foda — DISMISSED** [review]

**Review by jsuarezruiz — COMMENTED** [review]
LGTM but requires rebase.

**Review by PureWeen — CHANGES_REQUESTED** [review]
rebase isn't quite correct

**Review by PureWeen — APPROVED** [review]

**Comment by PureWeen** [comment]
- failing template test is unrelated

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue25038.cs (hunk 1 above)
Comment (NirmalKumarYuvaraj) [reply src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue25038.cs:22]: Thanks for the images

**Pull request merged**